### PR TITLE
fix: database prefix

### PR DESCRIPTION
**Pull request by radko93**

### CODE REVIEW

File: src/lib/interface.ts [hunk 1]
```diff
@@ -168,6 +168,9 @@ export class MobileDeviceInterface extends DeviceInterface {
 
   async getNamespacedKeychainValue(identifier: ApplicationIdentifier) {
     const keychain = await this.getRawKeychainValue();
+    if (identifier && identifier === 'standardnotes') {
```
Comment (mobitar) [inline src/lib/interface.ts:171]: This needs to be extracted into its own function. `isIdentifierLegacy`, with proper comments. Way too much repetition in this commit.

**Comment by radko93** [comment]
InstallationService does not work, seems like it needs to use raw keychain value always. If you remove data and create new application it won't recover that there is something in keychain.

File: src/lib/interface.ts [hunk 2]
```diff
@@ -249,4 +249,13 @@ export class MobileDeviceInterface extends DeviceInterface {
       })
       .catch(() => showAlert());
   }
+
+  /**
+   * We use this function to decide if we need to prefix the identifier in getDatabaseKeyPrefix or not.
+   * And to decide if the raw/namespaced keychain needs to be used.
+   * @param identifier The ApplicationIdentifier
+   */
+  private isLegacyIdentifier(identifier: ApplicationIdentifier) {
```
Comment (mobitar) [inline src/lib/interface.ts:258]: Should not be instance function. Should be generic JS utility function defined at top of file.

Two reasons for this:
1. This function has nothing to do with instances of this class, and its return value does not differ depending on the instance.
2. It communicates very clearly that this function doesn't do anything other than comparing const values.

File: src/lib/interface.ts [hunk 3]
```diff
@@ -249,4 +249,13 @@ export class MobileDeviceInterface extends DeviceInterface {
...
+    return identifier && identifier === 'standardnotes';
```
Comment (mobitar) [inline src/lib/interface.ts:259]: `'standardnotes'` should be extracted into a `const` variable, with a comment describing its significance and why it's equal to that particular value.

**Pull request merged**